aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,117 @@
 
-Learn about [handling exceptions during application startup](xref:fundamentals/error-handling#startup-exception-handling).
+[!code-csharp[Main](../common/samples/WebApplication1DotNetCore2.0App/Program.cs?name=snippet_Main&highlight=10)]
+
+The app can define separate `Startup` classes for different environments (for example, `StartupDevelopment`), and the appropriate startup class is selected at runtime by convention. The class whose name suffix matches the current environment is prioritized. If the app is run in the Development environment and includes both a `Startup` class and a `StartupDevelopment` class, the `StartupDevelopment` class is used. See [Working with multiple environments](xref:fundamentals/environments#startup-conventions) for more information.
+
+The `Startup` class constructor can accept dependencies defined by the host. A common use of DI with the `Startup` class is to inject [IHostingEnvironment](/dotnet/api/Microsoft.AspNetCore.Hosting.IHostingEnvironment). This approach can be used to configure services as an alternative to the convention-based approach described above:
+
+[!code-csharp[Main](startup/snapshot_sample/Startup2.cs)]
+
+To learn more about `WebHostBuilder`, see the [Hosting](xref:fundamentals/hosting) topic. For information on handling errors during startup, see [Startup exception handling](xref:fundamentals/error-handling#startup-exception-handling).
 
 ## The ConfigureServices method
 
-The [ConfigureServices](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.hosting.startupbase#Microsoft_AspNetCore_Hosting_StartupBase_ConfigureServices_Microsoft_Extensions_DependencyInjection_IServiceCollection_) method is optional; but if used, it's called before the `Configure` method by the web host. The web host may configure some services before ``Startup`` methods are called (see [hosting](xref:fundamentals/hosting)). By convention, [Configuration options](xref:fundamentals/configuration/index) are set in this method.
+The [ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configureservices) method, which is optional, is called by the web host before the `Configure` method to configure the app's services. The web host may configure some services before `Startup` methods are called (details are provided in the [Hosting](xref:fundamentals/hosting) topic). By convention, [Configuration options](xref:fundamentals/configuration/index) are set in the `ConfigureServices` method. Adding services to the service container makes them available within the app and in the `Configure` method resolved via [dependency injection](xref:fundamentals/dependency-injection) or from [IApplicationBuilder.ApplicationServices](/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder.applicationservices).
```

Comment:
>Adding services to the service container makes them available withi

Too long, too much going on.  Do you mean something like:

Adding services to the service container makes them available within the app via DI. They are available in config via IAppBuilder?

Reply:
We have to say that:
* This method is optional.
* That it's called before `Configure`.
* Web host config will already have added one or more services on its own before reaching this method.
* Config options are setup here (with a link over to the Config topic).

I don't think cutting that content is necessarily helpful; although, I admit that I may have been too economical squeezing it all in there.

Reply:
I like bullets. MT doesn't work with long/complex sentences.

Reply:
Sure ... I'll do that.